src/main/java/jarvis/storage/JsonAdaptedConsult.java

```diff
@@ -24,45 +30,88 @@ public class JsonAdaptedConsult extends JsonAdaptedLesson {
+        final LessonDesc modelLessonDesc = this.getLessonDesc() != null
+                ? new LessonDesc(this.getLessonDesc())
+                : null;
 
-        if (this.getTimePeriod() == null) {
+        // TimePeriod
+        if (this.getStartDateTime() == null || this.getEndDateTime() == null) {
             throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
                     TimePeriod.class.getSimpleName()));
         }
+        if (!TimePeriod.isValidTimePeriod(this.getStartDateTime(), this.getEndDateTime())) {
+            throw new IllegalValueException(TimePeriod.MESSAGE_CONSTRAINTS);
+        }
+        final TimePeriod modelTimePeriod = new TimePeriod(this.getStartDateTime(),
+                this.getEndDateTime());
 
+        // Student list for lesson
+        if (this.getStudentList() == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                    Student.class.getSimpleName()));
+        }
+        List<Student> modelStudentList = new ArrayList<>();
+        for (JsonAdaptedStudent jsonAdaptedStudent : this.getStudentList()) {
+            modelStudentList.add(jsonAdaptedStudent.toModelType());
+        }
+
+        // LessonAttendance
         if (this.getAttendance() == null) {
             throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
                     LessonAttendance.class.getSimpleName()));
         }
+        TreeMap<Student, Boolean> attendanceMap = new TreeMap<>(Comparator.comparing(s ->
+                s.getName().toString()));
+        for (Integer i : this.getAttendance().keySet()) {
+            attendanceMap.put(modelStudentList.get(i), this.getAttendance().get(i));
+        }
+        LessonAttendance modelAttendance = new LessonAttendance(attendanceMap);
 
-        if (this.getNotes() == null) {
+        // LessonNotes
+        if (this.getGeneralNotes() == null || this.getStudentNotes() == null) {
             throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
                     LessonNotes.class.getSimpleName()));
         }
+        TreeMap<Student, ArrayList<String>> modelStudentNotes = new TreeMap<>(Comparator.comparing(s ->
+                s.getName().toString()));
+        for (Integer i : this.getStudentNotes().keySet()) {
+            modelStudentNotes.put(modelStudentList.get(i), this.getStudentNotes().get(i));
+        }
+        LessonNotes modelLessonNotes = new LessonNotes(this.getGeneralNotes(), modelStudentNotes);
+
+
 
-        Consult consult = new Consult(modelLessonDesc, this.getTimePeriod().toModelType(),
-                new LessonAttendance(List.of(SampleStudentUtil.getSampleStudents())),
-                new LessonNotes(List.of(SampleStudentUtil.getSampleStudents())));
+        Consult consult = new Consult(modelLessonDesc, modelTimePeriod,
+                modelStudentList, modelAttendance, modelLessonNotes);
```

Comment:
+1, I think this will reduce a lot of the code duplication between the 3 Lesson subtypes since they contain similar components (attendance, notes etc.)